Lib/json/tool.py

```diff
@@ -68,12 +72,42 @@ def main():
             outfile = open(options.outfile, 'w', encoding='utf-8')
         with outfile:
             for obj in objs:
-                json.dump(obj, outfile, **dump_args)
+                if with_colors:
```

Comment:
> > What about CLIs? they would be broken if data is piped?
> 
> AFAIU they wouldn't (and local testing confirms that), see this comment from Hugo: [#131952 (comment)](https://github.com/python/cpython/issues/131952#issuecomment-2769303613)

By default it checks stdout. So this is fine and doesn't add colour:

```console
❯ cat in.json
{"first_name": "John"}

❯ ./python.exe -m json in.json | cat
{
    "first_name": "John"
}
```

But if you're outputting to something else, such as a an output file, it does include ANSI colour codes:

```console
❯ cat in.json
{"first_name": "John"}

❯ ./python.exe -m json in.json out.json

❯ cat out.json
{
    [38;2;248;248;240m[32m"first_name"[38;2;248;248;240m[0m: [38;2;248;248;240m[32m"John"[38;2;248;248;240m[0m
}
```

So let's make sure it checks the file we're using:

```suggestion
                if can_colorize(file=outfile):
```

```console
❯ ./python.exe -m json in.json out.json

❯ cat out.json
{
    "first_name": "John"
}
```